src/composable_kernel/composable_kernel/include/utility/static_buffer.hpp

```diff
@@ -0,0 +1,71 @@
+#ifndef CK_STATIC_BUFFER_HPP
+#define CK_STATIC_BUFFER_HPP
+
+#include "statically_indexed_array.hpp"
+
+namespace ck {
+
+template <AddressSpaceEnum_t BufferAddressSpace,
+          typename T,
+          index_t N,
+          bool InvalidElementUseNumericalZeroValue>
```

Comment:
Is this template parameter necessary since here we already have the second constructor ?

Reply:
Yes. Somehow, compiler needs compile-time constant ```InvalidElementUseNumericalZeroValue``` to determine whether to use the logic that return 0 or the logic that return customized value. Otherwise, I see a lot of scratch mem